pkg/collector/corechecks/system/disk/diskv2/disk.go

```diff
@@ -640,26 +633,42 @@ func (c *Check) fetchAllDeviceLabels() error {
 	return c.fetchAllDeviceLabelsFromBlkid()
 }
 
-// Factory creates a new check factory
-func Factory() option.Option[func() check.Check] {
-	return option.New(newCheck)
+// WithClock sets a custom clock on the Check and returns the updated Check.
+func (c *Check) WithClock(clock clock.Clock) *Check {
```

Comment:
We're adding more functions to the API just so that we can run tests from another package, that seems less and less convenient...

Reply:
Yeah but we're adding public APIs that are not used in the "real" code just so that we can test that, I don't see how that's better than just having the tests in the same package and setting the unexported fields in the tests

Reply:
I'm with @pgimalac on this one. The encapsulation is broken the moment we add an API that is only used, because we are not using the package in the test the way is going to be used in production (outside of the test package)

For this particular case, the fields are private, so they could only be modified inside the package, so being able to set them in the test seems less convoluted that the `.WithX()` pattern in this case

Reply:
Moved WithX methods from public API to some helpers into the `diskv2` package only available in tests.
This is a good solution after some research and combines the best of both worlds.